src/interchange/src/avro/schema.rs

```diff
@@ -448,8 +450,11 @@ impl SchemaCache {
                 // An issue with _fetching_ the schema should be returned
                 // immediately, and not cached, since it might get better on the
                 // next retry.
-                // TODO - some sort of exponential backoff or similar logic
-                let response = self.ccsr_client.get_schema_by_id(id).await?;
+                let ccsr_client = &self.ccsr_client;
+                let response = Retry::default()
+                    .max_duration(Duration::from_secs(30))
+                    .retry(|_| async move { ccsr_client.get_schema_by_id(id).await })
```

Comment:
I think that each failed attempt needs to be reported in the error log.

Reply:
What kind of error message would you propose? Given that this may be a transient error, I'm reluctant to add an error message. We do have an error message for when the fetch fails after the maximum time.

Is there something actionable for the user or would this be noise? Another option would be to add a Prometheus metric that tracks the number of retries and expose that metric?

Reply:
I think it should be a warning at least. This blocks avro decoding, so nothing in the dataflow can run while this is being retried. If someone's trying to debug why their queries are backing up, the fact that their schema registry configuration is flaky is something they'd surely be interested in knowing.

Reply:
I can add a warning message if we like and it'll probably look something like this:

```
warn: temporary failure trying to fetch schema by ID
warn: temporary failure trying to fetch schema by ID
error: permanent failure to fetch schema by ID
```

Given the 30 second timeout, I'm not sure how likely it will be that people can catch the warning messages and try to debug it before the error message pops up.

What do you think? I can add the warning messages if that makes sense!

Reply:
If the retry _succeeds_, you may never get an error message, right? I'm imagining a situation where someone's network is screwed up such that their schema registry sporadically goes down for 1 second at a time. You'd want to see that in the logs, even though there is no permanent error.

So, indeed, I think what you just described sounds good.

Reply:
Yes, if the retry succeeds, you'll never see an error message. But each retry will generate a warning, which can serve as a good "when does this issue arise" debugging hint.

Reply:
Sounds good!